pkg/grpc/endpoints.go

```diff
@@ -137,11 +134,40 @@ func setAPIEndpointsMiddlewares(rules []config.APIAccessRule) (grpc.UnaryServerI
 		func(srv any, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
 			// Apply the allowlist only on methods that are part of the Dapr runtime, or it will interfere with gRPC proxying
 			if strings.HasPrefix(info.FullMethod, daprRuntimePrefix) {
-				_, ok := allowed[info.FullMethod]
-				if !ok {
-					return invokev1.ErrorFromHTTPResponseCode(http.StatusNotImplemented, "requested endpoint is not available")
+				if len(allowed) > 0 {
+					_, ok := allowed[info.FullMethod]
+					if !ok {
+						return invokev1.ErrorFromHTTPResponseCode(http.StatusNotImplemented, "requested endpoint is not available")
+					}
+				}
+				if len(denied) > 0 {
+					_, ok := denied[info.FullMethod]
+					if ok {
+						return invokev1.ErrorFromHTTPResponseCode(http.StatusNotImplemented, "requested endpoint is not available")
+					}
```

Comment:
@ItalyPaleAle The order here is opposite of the order seeing in `http/endpoint.go`. In the HTTP one, denied is checked first and then allowed is checked. Here the opposite is true.